### ISSUE TITLE
apply classes instead of inline styles 

### ISSUE THREAD

**Issue by spooky** [isaacHagoel/svelte-dnd-action#84]
This is an improvement idea.
The `dropTargetStyle` is nice, but it may be more flexible to give the possibility to apply a custom class to the drop targets instead of inline styles.

**Comment by isaacHagoel** [comment]
this  feature can be added but how would it be more flexible? 
thx

**Comment by spooky** [comment]
styling would be in css instead where you have selector specificity, whereas currently, with inline styling, there's little room to override.

**Comment by chbert** [comment]
This would also be very helpful for using it in conjunction with tailwindcss (BTW: great library, thanks!)

**Comment by isaacHagoel** [comment]
@chbert any chance you create a demo REPL illustrating what you would like to be able to do?
i wanna to be sure Svelte's classes scoping wouldn't get in the way of something like what this issue requests.
Also need to remember that each dropzone could be a completely different component (that accepts similarly shaped data items).

**Comment by chbert** [comment]
@isaacHagoel I see your points, very valid! I think the idea would basically be, to be able to use style such as border-primary to style an element, to avoid having to manually enter colour values for example. But you are right about the scoping issues that would involve...

**Comment by chbert** [comment]
`dropTargetClasses={"border-primary"}`

**Comment by isaacHagoel** [comment]
Where would the class be defined and how (global?)?

**Comment by binary-koan** [comment]
I'm also using Tailwind CSS (so classes are defined globally outside Svelte). I tried implementing this and it doesn't work with styles in the Svelte `<style>` section though ... Svelte considers the class to be unused and deletes the definition during compilation.

I worked around it by tracking a boolean `isDragging` like this:

```
<script>
let isDragging = false
let items = [...]

function handleConsiderDrag(e) {
  items = ...
  isDragging = true
}

function handleFinalizeDrag(e) {
  items = ...
  isDragging = false
}
</script>

<div use:dndzone={{items}} on:consider={handleConsiderDrag} on:finalize={handleFinalizeDrag}>
  ...
</div>
```

It kinda works but it doesn't exactly match how the styles are added and removed because finalize isn't called until after the animation completes. So another idea might be to have `grab` and `release` events (not sure about the names) to do arbitrary custom logic at the moment of drag start / drop?

**Comment by isaacHagoel** [comment]
@binary-koan looks like the most common use case has to do with global classes (ex: tailwind).
this makes this quite easy to implement under a few assumptions (ex: that the class is not already on the dropzone before drag starts so it is safe to always remove it from the classList after drop).
maybe I can even name the option something like `dropTargetGlobalClasses` and take an array of strings.
makes sense?

**Comment by chbert** [comment]
@isaacHagoel like this, right?
 https://svelte.dev/repl/3a80822546c642388d328ca79ca225b5?version=3.22.3

I think that would be great! Not sure if the array is a technical requirement on your side? Otherwise, one string would be okay as well, like this:
https://svelte.dev/repl/df10a8411edd4cb4b6b71827ae59d741?version=3.22.3

JFYI Tailwind also allows for variants like `hover:border-red-300` (not sure if that changes anything though)

**Comment by isaacHagoel** [comment]
@chbert yeah something like that.
The list is not a hard requirement on my end I just thought it might be nicer to be more explicit and also it plays nice with `el.classList.add(...dropTargetClasses)`.
I will try to get it done this week.

**Comment by isaacHagoel** [comment]
this feature is available in version 0.8.1 and onwards. thanks everyone for the inputs
